### PR TITLE
Xinyu HW3

### PR DESCRIPTION
**Pull request by Wentworthliu123**

### CODE REVIEW

**Comment by dhruvalb** [comment]
Evaluation
----------------------------------------------------------------------------
| Topic					 	|Total	| Score | Comments |
|-----------------------------------------------|-------|-------|----------|
| Decision Trees				|	|	||
| **Create Training Set**			|10	|10	||
| **Plot training and test set** 		|15	|15	||
| **Update lambda, retest and discuss** 	|10	|10	||
| **Apply bagging to the training set** 	|10	|10	||
| **Apply random forest to training set**	|10	|10	||
| **Apply linear regression to training set** 	| 5	| 5	||
| **Compare test errors across all fits**	| 5	| 5	||
| Support Vector Machines			|	|	||
| **Fit a support vector classifier**		|10	|10	||
| **Display the confusion matrix**		| 5	| 2	| The question asks for error rate not accuracy. Points deducted here but penalty for the same thing is not applied in later question. |
| **Find an optimal cost**			|10	|10	||
| **Compute and discuss differences**		|10	|5	| Missing discussion of results|
| **TOTAL**            				|100	| 92	||
